core/src/main/java/org/elasticsearch/cluster/routing/IndexRoutingTable.java

```diff
@@ -128,6 +128,11 @@ boolean validate(MetaData metaData) {
                     throw new IllegalStateException("shard routing has an index [" + shardRouting.index() + "] that is different " +
                                                     "from the routing table");
                 }
+                if (shardRouting.active() &&
+                    indexMetaData.activeAllocationIds(shardRouting.id()).contains(shardRouting.allocationId().getId()) == false) {
+                        throw new IllegalStateException("active shard routing " + shardRouting + " has no corresponding entry in the " +
```

Comment:
`ShardRouting#toString()` prints a short summary, which is quite verbose for an exception message.  How about changing this to:

```
throw new IllegalStateException("active shard routing " + shardRouting.shardId() + " has allocation id [" + shardRouting.allocationId().getId() + "] which has no corresponding entry in the in-sync allocation set " + indexMetaData.activeAllocationIds(shardRouting.id()));
```

or something like that